modules/programs/neovim.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.neovim;
+in
+{
+  options = {
+    programs.neovim = {
+      enable = mkEnableOption "Neovim";
+
+      withPython = mkOption {
+        type = types.nullOr types.bool;
+        default = true;
+        description = ''
+          Enable python2 provider. Set to true to use python2-based plugins
+        '';
+      };
+
+      extraPythonPackages = mkOption {
+        type = types.listOf types.package;
+        default = [ ];
+        example = [ ];
+        description = ''
+          List here python2 packages required for your plugins to work.
+        '';
+      };
+
+      withRuby = mkOption {
+        type = types.nullOr types.bool;
+        default = true;
+        description = ''
+          Enable ruby provider.
+        '';
+      };
+
+      withPython3 = mkOption {
+        type = types.nullOr types.bool;
+        default = true;
+        description = ''
+          Enable python3 provider. Set to true to use python"-based plugins
```

Comment:
The `python"-based` part seems to have some stray " character.

Reply:
sorry, it happens when switching layouts :'(